crates/swc_ecma_transforms_module/tests/fixture/common/misc/import-const-throw/output.amd.js

```diff
@@ -0,0 +1,23 @@
+define([
+    "require",
+    "exports",
+    "foo",
+    "bar",
+    "baz"
+], function(require, exports, _foo, _bar, _baz) {
+    "use strict";
+    Object.defineProperty(exports, "__esModule", {
+        value: true
+    });
+    _foo = _interopRequireDefault(_foo);
+    _bar = _interopRequireWildcard(_bar);
+    Foo = 42;
```

Comment:
I think this will not throw an error if we don't have `use strict`.
But I think the output should throw regardless of `strict` if original source code throws

Reply:
I moved the reassign check into the lint pass.
- #5033